### ISSUE TITLE
Someone Shown As the Post Author

### ISSUE THREAD

**Issue by TheJeffr0** [Automattic/jetpack#3028]
Ok, the title of this issue is weird but I had a hard time coming up with something else. This occurs in Jetpack Notifications and also on the WordPress for iOS Mobile App. When a person leaves a parent comment on an article, Someone shows up as the post author. This screenshot does a better job of explaining. Where it says Someone should actually be the Post Author. 
![notificationssomeone](https://cloud.githubusercontent.com/assets/6315977/11215821/2659b5f4-8d16-11e5-9193-88216f73288c.png)

**Comment by kraftbj** [comment]
For the post author of that post, is their local site's account linked to their WP.com account?

**Comment by TheJeffr0** [comment]
I'm the author of that post and the Tavern is linked to my WP.com account.

**Comment by TheJeffr0** [comment]
The site this is happening on is using WordPress 4.4 beta 4.

**Comment by kraftbj** [comment]
Confirmed on my test site too. Post author info doesn't seem to be passed to Notif. @dmsnell something else for your collection? :)

**Comment by dmsnell** [comment]
Usually this means that the author information isn't available. @kraftbj can you go one step further and double check that not only is the Jetpack site linked via the WordPress.com user, but also that the Jetpack user itself is connected to a WordPress.com user?

**Comment by kraftbj** [comment]
@dmsnell Indeed. Local user is the master user for the JP connection, so they are.

**Comment by jeherve** [comment]
Also in 2537591-t and p2XLIi-1gG-p2

**Comment by rralian** [comment]
@dmsnell is this active? Looks like we're still getting reports.

**Comment by dmsnell** [comment]
I haven't touched this code so I'm sure it's still an issue. We can probably start to think about fixing it. It's probably going to show up always if there is any kind of issue with the connected user. My guess is that it tries getting user data but for some reason we don't have knowledge of that user in the WordPress.com servers.

If we can reproduce this reliably in a test Jetpack site that would be helpful. I can't promise any particular speed on the bug but I'll try!
